### PR TITLE
Add watch() API

### PR DESCRIPTION
**Pull request by kenotron**
## Overview

Adding `glob-watcher` (another gem from the gulp ecosystem) to allow watched task functions. Whenever a file changed in the glob, it'll trigger the task function.

1. `just-task`'s `watch()` a watcher, they can participate in parallel / series composition with `() => watch`
2. watch will take on sync taskfunction's like other `just-task` functions

### CODE REVIEW

File: packages/just-task/package.json [hunk 1]
```diff
@@ -28,10 +28,13 @@
     "resolve": "^1.19.0",
     "undertaker": "^1.3.0",
     "undertaker-registry": "^1.0.1",
-    "yargs-parser": "^20.2.3"
+    "yargs-parser": "^20.2.3",
```
Comment (dzearing) [inline packages/just-task/package.json:31]: Shouldn't `chokidar` be in the dependencies here?

Comment (kenotron) [reply packages/just-task/package.json:31]: chokidar is a dep in glob-watcher not here.

Comment (kenotron) [reply packages/just-task/package.json:31]: I'm using @types/chokidar, not chokidar. Basically only importing the watcher types.

Comment (kenotron) [reply packages/just-task/package.json:31]: chokidar itself is a dep of glob-watcher...

Comment (dzearing) [reply packages/just-task/package.json:31]: oh got it, i see now.

**Pull request merged**